File: sources/foo/Foo.m
//  Copyright (c) 2017 Milen Dzhumerov. All rights reserved.

#import "Foo.h"

#import "Foo-Swift.h"

@implementation FooObject

- (id)makeBaz {
	// Obj-C -> Swift
  return [BazObject new];
}

@end

File: sources/Foo/Foo.swift
//  Copyright (c) 2017 Milen Dzhumerov. All rights reserved.

import Foundation

// Swift-only Types

public struct FooStruct {
	let name: String
}

// Obj-C Types

public class FooLogger: NSObject {
  public func log() {
		// Swift -> Obj-C
    let foo = FooObject()
    print("foo: \(foo.makeBaz())")
  }
}

@objc public class FooSubclass: FooObject {
  
}
